model_signing/hashing/file.py

```diff
@@ -219,4 +225,4 @@ def compute(self) -> hashing.Digest:
     def digest_name(self) -> str:
         if self._digest_name_override is not None:
             return self._digest_name_override
-        return f"file-{self._content_hasher.digest_name}-{self._shard_size}"
+        return f"file-{self._content_hasher.digest_name}-{self.shard_size}"
```

Comment:
note: here we have to be careful that digest name do not contain the character `-`. Either we add a check somewhere, or we use a different character, like `$`, `|`. Maybe you're thinking of another PR to customize this value anyway, since `name-shard_size` could lead to confusion anyway since the sharding "algo" is not specified

Reply:
Hmm, we could standardize on using `.` everywhere between the fields.

So far, we control all the components so there's no risk of confusion. I'm pushing testing /more customization for this after we have most of the API in place (at least after we have the full manifest format), as we should be able to quickly fix issues.